OSDOCS-9437: adds audit log config policies MicroShift (#75233)
openshift/openshift-docs: modules/microshift-audit-logs-config-proc.adoc

    maxFiles: 1 <2>
    maxFileAge: 7 <3>
    profile: Default <4>
----
<1> The maximum audit log file size in megabytes. If the value is 0, the limit is disabled. In this example, if the live log reaches the 200Mb limit, it is rotated, causing any existing log backup to be deleted.
<2> The maximum number of rotated audit log files to retain. After the limit is reached, the log files in order from oldest to newest are deleted until the specified limits are reached. When the value is 0, the limit is disabled. In this example, the value `1` results in only 1 file of size `maxFileSize` being retained in addition to the current active log.
<3> Specifies the maximum time in days that log files are kept. Files older than this limit will be deleted. When the value is 0, the limit is disabled. In this example, after a log file is more than 7 days old, it is deleted. The deletion happens regardless of whether the live log has reached the maximum file size as give in the `maxFileSize` field value.
<4> Logs only metadata for read and write requests; does not log request bodies except for OAuth access token requests. If you do not specify this field, the Default profile is used.

> . Optional: To specify a specific directory for logs, move the `/var/log/kube-apiserver` directory to your desired location by running the following command:

Reviewer: ```suggestion
. Optional: To specify a specific directory for logs, first stop MicroShift, then move the `/var/log/kube-apiserver` directory to your desired location by running the following command:
```

// Text snippet included in the following assemblies:
//
// * microshift_configuring/microshift-audit-logs-config.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-configuring-audit-log-values_{context}"]
= Configuring audit log values

You can configure audit log settings by using the {microshift-short} service configuration file.

.Procedure

> . Make a copy of the provided `config.yaml.default` file in the `/etc/microshift/` directory, renaming it `config.yaml`. Keep the new {microshift-short} `config.yaml` you create in the `/etc/microshift/` directory. The new `config.yaml` is read at system start and takes precedence over built-in settings.

Author: Yes, it means whenever the MicroShift starts, either for the first time or as a restart.

// Text snippet included in the following assemblies:
//
// * microshift_configuring/microshift-audit-logs-config.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-configuring-audit-log-values_{context}"]
= Configuring audit log values

> Use the {microshift-short} service configuration file to implement custom audit log settings.

Reviewer 2: ```suggestion
By using the {microshift-short} service configuration file, you can implement custom settings for the audit log.
```

Author: Better: You can configure audit log settings by using the {microshift-short} service configuration file.

.Procedure

> . Add the following `apiServer` stanza to your {microshift-short} `config.yaml` file. If you need to make the file, run the following command:

Reviewer 2: ```suggestion
. To make the file, run the following command:
```

+
[source,terminal]
----
$ sudo cat /etc/microshift/config.yaml
----

> +

Reviewer 2: ```suggestion
. Add the following content to your {microshift-short} `config.yaml` file:
+
```

.Example `auditLog` configuration
[source,yaml]
----
apiServer:
# ....
  auditLog:
    maxFileSize: 200 # <1>
    maxFiles: 1 # <2>
    maxFileAge: 7 # <3>
    profile: Default # <4>
# ....
----

> <1> The maximum audit log file size in megabytes. If the value is 0, the limit is disabled. In this example, after the live log reaches the 200Mb limit, it is rotated.
> <2> The maximum number of rotated audit log files to retain. After the limit is reached, the log files in order from oldest to newest are deleted until the specified limits are reached. When the value is 0, the limit is disabled. In this example, the value `1` results in only 1 file of size `maxFileSize` being retained in addition to the current active log.
> <3> Specifies the maximum time in days that log files are kept. Files older than this limit are deleted. When the value is 0, the limit is disabled. In this example, after a log file is more than 7 days old, it is deleted. The deletion happens regardless of whether the live log has reached the maximum file size as give in the `maxFileSize` field value. File age is determined by the timestamp written in the name of the rotated log file, for example, `audit-2024-05-16T17-03-59.994.log`.
> <4> Logs only metadata for read and write requests; does not log request bodies except for OAuth access token requests. If you do not specify this field, the Default profile is used.

Reviewer 2: ```suggestion
<1> The maximum size in megabytes for the audit log file. If the value is `0`, the limit is disabled. In this example, the file is rotated as soon as the live log reaches the 200 MB limit.
<2> The maximum number of rotated audit log files that you want to retain. After the limit is reached, the log files are deleted in order from oldest to newest until the specified limit is reached. If the value is `0`, the limit is disabled. In this example, the value `1` means that only 1 file of the `maxFileSize" is retained in addition to the current active log.
<3> Specifies the maximum time in days that log files are kept. Files older than this limit are deleted. If the value is `0`, the limit is disabled. In this example, a log file that is older than 7 days is deleted. The deletion takes place regardless of whether the live log has reached the maximum file size specified in the `maxFileSize` field. The file age is determined by the timestamp in the name of the rotated log file, for example, `audit-2024-05-16T17-03-59.994.log`.
<4> Logs only metadata for read and write requests; does not log request bodies, except for OAuth access token requests. If you do not specify this field, the default profile is used.
```

> . Optional: To specify a specific directory for logs, you can stop {microshift-short}, then move the `/var/log/kube-apiserver` directory to your desired location:

Reviewer 2: ```suggestion
. Optional: To specify a specific directory for logs, you can stop {microshift-short}, and then move the `/var/log/kube-apiserver` directory to your desired location:
```

.. Stop {microshift-short} by running the following command:
+
[source,terminal]
----
$ sudo systemctl stop microshift
----
.. Move the `/var/log/kube-apiserver` directory to your desired location by running the following command:
+
[source,terminal]
----
$ sudo mv /var/log/kube-apiserver <~/kube-apiserver> <1>
----

> <1> Replace _<~/kube-apiserver>_ with the path to the directory that you want to use.

Reviewer 2: ```suggestion
<1> Specify the path to the directory that you want to use.
```

> . If you specified a specific directory for logs, create a symlink to your custom directory at `/var/log/kube-apiserver` by running the following command:

Reviewer 2: Optional ??

Author: Not _really_. We want logs to be accessible to sos reports and this is how we achieve that.

+
[source,terminal]
----
$ sudo ln -s <~/kube-apiserver> /var/log/kube-apiserver <1>
----
<1> Replace _<~/kube-apiserver>_ with the path to the directory that you want to use. This enables the logs to be collected in sos reports.

> . If you are configuring audit log policies on a running instance, restart {microshift-short} by entering the following command:

Reviewer 2: Optional ??
